core/src/main/java/io/questdb/network/Net.java

```diff
@@ -279,9 +283,18 @@ public static long sockaddr(int ipv4address, int port) {
     }
 
     public static int socketTcp(boolean blocking) {
-        return Files.bumpFileCount(socketTcp0(blocking));
+        return Files.bumpFileCount(setKeepAlive(socketTcp0(blocking), TCP_KEEPALIVE_SECONDS));
```

Comment:
This call chain doesn't look nice. If `socketTcp0()` is successful and `setKeepAlive()` fails, we'll return a negative value and leak the fd.

Better to keep them separate and maybe log something if `setKeepAlive()` fails instead of failing the fd creation.

Reply:
> we'll return a negative value and leak the fd. 

that's a good point, thanks!

Reply:
how about now?